### ISSUE TITLE
Filter by _not_ in cohort

### ISSUE THREAD

**Issue by macobo** [PostHog/posthog#2273]
## Is your feature request related to a problem?

I want to track feature X, but usage stats include both internal users as well as real users.

## Describe the solution you'd like

Allow filtering by NOT in cohort.

## Additional context

Example: Track how many times session recording is viewed.

**Comment by theonlyjohnny** [comment]
Is there any timeline on this issue?

**Comment by samwinslow** [comment]
Should the logic here be:

1. User is in any cohort that is not `X` or user cohort is unset
2. User is in any cohort that is not `X` and user cohort is set

I would assume the first, but the distinction seems important.

**Comment by macobo** [comment]
It is 3. Assume user is in cohorts `C = [1,2,3,...]` (can be an empty list). User is not in cohort A if `A not in C`. :)

**Comment by macobo** [comment]
Duplicates: https://github.com/PostHog/posthog/issues/2978 https://github.com/PostHog/posthog.com/issues/1157 and https://github.com/PostHog/posthog/issues/3298

cc @clarkus I think design here is a minor blocker. Worth thinking about this in the backburner?

**Comment by clarkus** [comment]
Updated filter concepts that allow for specifying in cohort / not in cohort within a given time range. Note this also encompasses work from https://github.com/PostHog/posthog/issues/2594. 

![Filters](https://user-images.githubusercontent.com/254612/144307313-96580f42-bdfc-46e7-9008-7be90ee2fe1a.png)
https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf/PostHog-App?node-id=5589%3A33881

**Comment by macobo** [comment]
In cohort/not in cohort with a time range filter doesn't match the semantics we have for cohorts.

**Comment by clarkus** [comment]
> In cohort/not in cohort with a time range filter doesn't match the semantics we have for cohorts.

Got it so time ranges are not applicable to those filters.

**Comment by EDsCODE** [comment]
Implemented in new cohort filtering pattern https://github.com/PostHog/posthog/pull/10049 https://github.com/PostHog/posthog/pull/9595

**Comment by Twixes** [comment]
I don't think this is actually solved yet, because you need to create a special "Not Some Other Cohort" for every "Some Other Cohort" to actually use negative cohort filtering in insights. (E.g. see this internal feedback ticket: ZEN-2838)

**Comment by mariusandra** [comment]
This is now solved with HogQL: https://github.com/PostHog/posthog/pull/16119

You can do `and person_id not in cohort x`
